crates/nu-command/src/platform/input/input_.rs

```diff
@@ -75,139 +80,97 @@ impl Command for Input {
+                return Err(ShellError::IOError(
                     "input can't stop on this byte".to_string(),
-                ))
+                ));
             }
         } else {
-            if let Some(prompt) = prompt {
-                print!("{prompt}");
-                let _ = std::io::stdout().flush();
-            }
+            None
+        };
 
-            let mut buf = String::new();
+        if let Some(prompt) = prompt {
+            print!("{prompt}");
+            let _ = std::io::stdout().flush();
+        }
 
-            if suppress_output || numchar_exists {
-                crossterm::terminal::enable_raw_mode()?;
-                // clear terminal events
-                while crossterm::event::poll(Duration::from_secs(0))? {
-                    // If there's an event, read it to remove it from the queue
-                    let _ = crossterm::event::read()?;
-                }
+        let mut buf = String::new();
 
-                loop {
-                    if i64::try_from(buf.len()).unwrap_or(0) >= numchar.item {
-                        let _ = crossterm::terminal::disable_raw_mode();
-                        break;
-                    }
-                    match crossterm::event::read() {
-                        Ok(Event::Key(k)) => match k.kind {
-                            KeyEventKind::Press | KeyEventKind::Repeat => {
-                                match k.code {
-                                    // TODO: maintain keycode parity with existing command
-                                    KeyCode::Char(c) => {
-                                        if k.modifiers == KeyModifiers::ALT
-                                            || k.modifiers == KeyModifiers::CONTROL
-                                        {
-                                            if k.modifiers == KeyModifiers::CONTROL && c == 'c' {
-                                                crossterm::terminal::disable_raw_mode()?;
-                                                return Err(ShellError::IOError(
-                                                    "SIGINT".to_string(),
-                                                ));
-                                            }
-                                            continue;
-                                        }
-
-                                        buf.push(c);
-                                    }
-                                    KeyCode::Backspace => {
-                                        let _ = buf.pop();
+        crossterm::terminal::enable_raw_mode()?;
+        // clear terminal events
+        while crossterm::event::poll(Duration::from_secs(0))? {
+            // If there's an event, read it to remove it from the queue
+            let _ = crossterm::event::read()?;
+        }
+
+        loop {
+            if i64::try_from(buf.len()).unwrap_or(0) >= numchar.item {
+                let _ = crossterm::terminal::disable_raw_mode();
```

Comment:
For breaking out of the loop without error you don't need to disable the raw mode twice, right?

```suggestion
```

Reply:
https://github.com/nushell/nushell/pull/10150/commits/774dae7b94732f17861ce7a070a5f04e7e8f5a48